io/io/src/TBufferJSON.cxx

```diff
@@ -570,6 +682,15 @@ TString TBufferJSON::ConvertToJSON(const void *obj, const TClass *cl, Int_t comp
       return TBufferJSON::ConvertToJSON(ptr, member, compact, arraylen);
    }
 
+   if (cl && cl->GetCollectionProxy() && dynamic_cast<TEmulatedCollectionProxy*>(cl->GetCollectionProxy())) {
+      ::Error("TBufferJSON::ConvertToJSON",
+            "The class requested (%s)"
+            " is an instance of an stl collection and does not have a compiled CollectionProxy."
+            " Please generate the dictionary for this collection (%s). No data will be written.",
+            cl->GetName(), cl->GetName());
+      return TString();
+   }
+
    TBufferJSON buf;
```

Comment:
Hi Philippe @pcanal 

Is this check correct? I copied it from TDirectory code.
Here I want to be sure that proxy exists for STL classes. 
Seems to be, it also resolves problem with following macro:
```
std::map<std::string,int> obj;
obj["name"] = 10;
auto json = TBufferJSON::ToJSON(&obj);
```
Before it was not working - now it does.